skimage/feature/_texture.pyx

```diff
@@ -43,31 +46,36 @@ def _glcm_loop(any_int[:, ::1] image, cnp.float64_t[:] distances,
         Py_ssize_t a_idx, d_idx, r, c, rows, cols, row, col, start_row,\
                    end_row, start_col, end_col, offset_row, offset_col
         any_int i, j
-        cnp.float64_t angle, distance
+        cnp.float64_t angle, distance, sin_angle, cos_angle
 
     with nogil:
         rows = image.shape[0]
         cols = image.shape[1]
 
         for a_idx in range(angles.shape[0]):
             angle = angles[a_idx]
+            sin_angle = sin(angle)
+            cos_angle = cos(angle)
+
             for d_idx in range(distances.shape[0]):
                 distance = distances[d_idx]
-                offset_row = round(sin(angle) * distance)
-                offset_col = round(cos(angle) * distance)
+                offset_row = round(sin_angle * distance)
+                offset_col = round(cos_angle * distance)
                 start_row = max(0, -offset_row)
                 end_row = min(rows, rows - offset_row)
                 start_col = max(0, -offset_col)
                 end_col = min(cols, cols - offset_col)
                 for r in range(start_row, end_row):
                     for c in range(start_col, end_col):
-                        i = image[r, c]
-                        # compute the location of the offset pixel
-                        row = r + offset_row
-                        col = c + offset_col
-                        j = image[row, col]
-                        if 0 <= i < levels and 0 <= j < levels:
-                            out[i, j, d_idx, a_idx] += 1
+                        if mask[r, c]:
+                            row = r + offset_row
+                            col = c + offset_col
+                            if mask[row, col]:
+                                i = image[r, c]
+                                j = image[row, col]
+
+                                if 0 <= i < levels and 0 <= j < levels:
```

Comment:
Thanks for checking the performance cost of this new PR. I can confirm an increase locally which isn't ideal. That got the thinking. 

Could we use this check for `0 <= i` and `0 <= j` to avoid the additional `mask[r, c]` check? If a `mask` is given we set `levels` at the appropriate positions to `0`. That way we'd have the same amount of checks in the hot loop. But we trade that for code clarity.

Let me know what you think. :)

Reply:
This works like a charm. Got these runtimes.

```
Scikit-image version: 0.24.0
Average execution time over 500 runs: 0.003027 seconds

Scikit-image version: 0.24.1rc0.dev0
Average execution time over 500 runs (mask=None): 0.002923 seconds
Average execution time over 500 runs (mask=ones): 0.002900 seconds
```

I suspect the improved runtimes are from moving the computation of the sin and cos outside of the loop. Not a huge difference, but it's still something.

Thanks again :)